Mathlib/AlgebraicGeometry/EllipticCurve/Weierstrass.lean

```diff
@@ -542,6 +542,26 @@ variable {R : Type u} [CommRing R] (E : EllipticCurve R)
 def j : R :=
   E.Δ'⁻¹ * E.c₄ ^ 3
 
+theorem j_eq_zero_iff' : E.j = 0 ↔ E.c₄ ^ 3 = 0 := by
+  simp only [EllipticCurve.j, Units.mul_right_eq_zero]
+
+alias ⟨_, j_eq_zero'⟩ := j_eq_zero_iff'
+
+theorem j_eq_zero_iff [IsReduced R] : E.j = 0 ↔ E.c₄ = 0 := by
```

Comment:
Is there a scenario where we have a reduced ring but not an integral domain i.e. `NoZeroDivisors R`? I know that this is more general, but the proof is so much simpler with `NoZeroDivisors R`:
```
  rw [j_eq_zero_iff', pow_eq_zero_iff three_ne_zero]
```

Reply:
Should we generalize `pow_eq_zero_iff` from `NoZeroDivisors` to `IsReduced`?

Reply:
This sounds like a good idea!

Reply:
> Is there a scenario where we have a reduced ring but not an integral domain i.e. `NoZeroDivisors R`?

I don't know, an artificial example is `R[X,Y]/(X,Y)`, but I don't know an example to consider elliptic curves on it.

> Should we generalize `pow_eq_zero_iff` from `NoZeroDivisors` to `IsReduced`?

Problem: `IsReduced` is not imported in that file.

Reply:
> Can we add `IsReduced.pow_(eq|ne)_zero[_iff]` in `Mathlib.RingTheory.Nilpotent.Defs` in this PR? I'm happy with this.

Sure.

Reply:
Added.